File: 2597-the-number-of-beautiful-subsets/2597-the-number-of-beautiful-subsets.java
class Solution {
    private int result;

    private void backtrack(int idx, List<Integer> sub, int[] nums, int k) {
        if (idx == nums.length) {
            if (!sub.isEmpty()) result++;
            return;
        }

        if (!sub.contains(nums[idx] - k)) {
            sub.add(nums[idx]);
            backtrack(idx + 1, sub, nums, k);
            sub.remove(sub.size() - 1);
        }

        backtrack(idx + 1, sub, nums, k);
    }

    public int beautifulSubsets(int[] nums, int k) {
        Arrays.sort(nums);
        result = 0;
        backtrack(0, new ArrayList<>(), nums, k);
        return result;
    }
}

File: 2597-the-number-of-beautiful-subsets/2597-the-number-of-beautiful-subsets.py
class Solution:
    def beautifulSubsets(self, nums: List[int], k: int) -> int:
        def backtrack(idx, sub):
            if sub:
                nonlocal result
                result += 1

            if idx == len(nums):
                return

            for i in range(idx, len(nums)):
                flag = False
                for n in sub:
                    if abs(nums[i] - n) == k:
                        flag = True
                        break
                if flag: continue
                sub.append(nums[i])
                backtrack(i + 1, sub)
                sub.pop()


        result = 0
        backtrack(0, list())
        return result